tests/test_api.py

```diff
@@ -482,14 +496,23 @@ def test_linked_thumbnail_url(self):
         path = os.path.abspath(os.path.expanduser(
             os.path.join(this_dir, "sg_logo.jpg")))
 
-        thumb_id = self.sg.upload_thumbnail("Project",
-            self.version['project']['id'], path)
+        thumb_id = self.sg.upload_thumbnail("Project", self.version['project']['id'], path)
 
+        attempts = 0
         response_version_with_project = self.sg.find(
             'Version',
             [['id', 'is', self.version['id']]],
             fields=['id', 'code', 'project.Project.image']
         )
+        while (attempts < THUMBNAIL_MAX_ATTEMPTS and
```

Comment:
Since in this query we're asking for `Version` entities but getting the `project.Project.image`, I needed to modify the await thumbnail method a bit, but I added an additional kwarg and subbed it in.